applications/nftaa.md

```diff
@@ -0,0 +1,411 @@
+
+Differences and Distinctions (NFTAA vs. EIP-6551):
+
+NFTAA is designed specifically for the Polkadot and Substrate ecosystems, utilizing their unique features, such as interoperability and cross-chain communication.
+Additionally, NFTAA’s focus on using NFTs as proxy accounts for staking and asset management offers a novel approach compared to the more traditional liquid staking and smart contract enhancements in the Ethereum ecosystem.
+
+EIP 6551 presents certain limitations. Our solution, e.g., does not prioritize connecting existing NFTs but enables bidirectional binding of the NFT and proxy.
+Consequently, upon ownership of an NFT, we can discern from the NFT metadata indicating its status as an NFTAA.
+Furthermore, we enhance transparency by ensuring the atomic creation of the NFT and proxy within a single transaction, thereby promoting seamless integration.
+Likewise, our design does not need the registrar service to create the TBA address as opposed to TBA.
+In our case, the pallet creates the NFTAA with functionality that also works as the NFTAA address (TBA uses one more address than NFTAA).
+Another difference is that in TBA, you could have more accounts on one NFT, which, in our case, is not possible due to the atomic creation of exactly one NFT at the time of creating the NFTAA.
+Regarding security, there are at least two differences between NFTAA and TBA. The first is the possibility of generating the address of TBA sooner without having the deployed contract (as the create2 function for creating contracts is deterministic, i.e., you can calculate the future contract address).
+In this case, if the upcoming smart contract is badly developed (does not contain an execute function for byte calls to other contracts) or is not upgradeable, it can happen that assets or liquidity sent to the TBA will not be withdrawable and become locked forever (even the NFT bounded in TBA can be sent by a mistake (if not safeguarded) to the address of TBA. Therefore you lock yourself from it).
+The problem is that TBA uses only the address of any contract you want in the registry contract and does not look for any details on how the contract looks or if it is already deployed. In our case, the account form is pre-defined in the factory pallet, and even after deployment, we guarantee upgradeability as it is a proxy account.
+The second security issue is potential fraud, which is [described in the EIP 6551 specification](https://eips.ethereum.org/EIPS/eip-6551\#fraud-prevention) that you withdraw assets from the TBA and at the same time sell the NFT to someone who thought that they would also have the assets in the TBA. In our case, we provide the security measures for not withdrawing the assets from the NFTAA while also selling the NFT, which is bound to the NFTAA.
```

Comment:
Will these security measures be included in your currently developed pallet? I don't see it in the milestones.

Reply:
We will try to address this problem within existing features, but at this point we cannot exclude the possibility that this mechanism may need to be improved later.

Reply:
for example, one possibility is: If NFTAA is listed you will not be able to sell assets owned by NFTAA

Reply:
That sounds good 👍  I'd like to see that. Would you be willing to integrate this into the deliverables?

Reply:
Ofc, I added it to [milestone 1](https://github.com/w3f/Grants-Program/blob/4bdd90ff5dc30e1d4db13d0701df2568c1c773bc/applications/nftaa.md#milestone-1--basic-functionality) in point 1d